### ISSUE TITLE
Templates Blanked or Deleted when Making Changes to Global Styles with Gutenberg Active

### ISSUE THREAD

**Issue by masperber** [Automattic/wp-calypso#92191]
### Quick summary

Under certain circumstances, a template can be blanked out when making changes to Global Styles, if Gutenberg is active. This "blanking out" appears a revision entry, and it can be reversed by visiting the template's revision history, but from the user's perspective, it is difficult to understand what has happened or how to reverse it.

### Steps to reproduce

This is one set of steps to reproduce this issue. There may be other circumstances that can produce the bug.

1. Create a plugin-enabled site on WordPress.com
2. Activate the theme Zaino
3. Ensure that WooCommerce and Gutenberg plugins are active
4. Go to Appearance > Editor, then Templates, then edit the template Page: Cart
5. Open Global Styles
6. Click on Blocks, then search for the Mini-Cart Proceed to Checkout Button
7. Change the button text to another color
8. Note that the "Are you ready to save?" prompt indicate a change to the Page: Cart template as well as Custom Styles. Unchecking the "Page: Cart" template change will prevent the bug from occurring. Leave it checked to verify the bug.
9. On saving, the Pattern selection modal will appear.
10. Visit {siteurl}/cart. Note that the page is completely blank.

### What you expected to happen

Making a change in Global Styles should not make any change to the template that is currently being edited.

### What actually happened

The edited template is blanked out.

### Impact

All

### Available workarounds?

Yes, difficult to implement

### Platform (Simple and/or Atomic)

Atomic

**Comment by masperber** [comment]
Initial report: 8392059-zen

Customer requests followup on resolution.

**Comment by mrfoxtalbot** [comment]
Thank you for the detailed report @masperber. It sounds like the likely culprit here is the Mini-Cart block, right?

Were you able to replicate this using a different theme? 

@Aljullu, are you still working on Woo blocks? If not, who could help triage this? Thank you!

**Comment by dsas** [comment]
There's some investigation done on this here: p1719520111073669-slack-C02FMH4G8 and https://github.com/Automattic/wp-calypso/issues/91763 has the exact same symptoms

**Comment by Aljullu** [comment]
> @Aljullu, are you still working on Woo blocks? If not, who could help triage this? Thank you!

Thanks for the ping, @mrfoxtalbot!

> Note that the "Are you ready to save?" prompt indicate a change to the Page: Cart template as well as Custom Styles.

I was able to replicate the `Page: Cart` template appearing in the save sidebar unnecessarily, but upon saving nothing weird happened. The template was still there and the frontend still worked.

Pinging Rubik's porter in case he can give it a try as well. cc @mikejolley

**Comment by Aljullu** [comment]
> I was able to replicate the `Page: Cart` template appearing in the save sidebar unnecessarily

I opened an issue in the WooCommerce repo about this: https://github.com/woocommerce/woocommerce/issues/48936.

**Comment by dsas** [comment]
The takeaway from the slack thread so far is this:
> so far I've only managed to reproduce it on an Atomic site with Zaino, when editing the Cart template and when using the undo button to save more than 2 entities (e.g. template, GS, and footer).
> 
> any other combination doesn't trigger the issue for me.
>
> it sounds like a race condition somewhere when saving multiple entities at the same time

It's also reproducible in wp studio, so it's not some dotcom specific thing.

**Comment by Aljullu** [comment]
Some questions that would be good to know:

1. Is this reproducible with other WooCommerce templates? Ie: Product Catalog, Single Product, etc. or only with Page: Cart?
2. Is this reproducible with non-WooCommerce templates? Ie: Blog Home.
3. Can the issue be reproduced consistently or it happens every other time you try?

In Slack I see some folks mentioned that pressing _Undo_ allows reproducing the issue, but it didn't work for me either. :confused: 

cc @roykho as well, as the porter of Origami.

**Comment by mikejolley** [comment]
I was able to replicate @Aljullu I'll comment on the issue in woo repository.

**Comment by dsas** [comment]
> Is this reproducible with other WooCommerce templates? Ie: Product Catalog, Single Product, etc. or only with Page: Cart?

I've also repro'd it on the `page-checkout` woo template. I haven't reprod it in Single Product template

> Is this reproducible with non-WooCommerce templates? Ie: Blog Home.

I haven't managed to repro it on non-woo templates yet, and neither has any of the other investigators in the slack thread afaik. 

> Can the issue be reproduced consistently or it happens every other time you try?

Consistently. Here's a video taken first time of asking

https://github.com/Automattic/wp-calypso/assets/93301/b0b28031-05a9-432d-b07d-f0f207f21405

**Comment by supernovia** [comment]
Quite possibly another case here; their home page template keeps getting blanked:
8513169-zd-a8c

**Comment by annezazu** [comment]
Is this still a problem? I saw this and it felt incredibly similar to another reported issue: https://github.com/Automattic/wp-calypso/issues/93471 cc @Aljullu @mikejolley @dsas (since you commented on this prior issue too but didn't connect to this).

It looks like new reports haven't come in for a few months either. I tried repeatedly to replicate on a self hosted site and wasn't able to. If the problem can be replicated, I'd like to get it fixed ASAP.
